common/network-common/src/main/java/org/apache/spark/network/util/TransportFrameDecoder.java

```diff
@@ -139,14 +139,18 @@ private ByteBuf decodeNext() throws Exception {
       return nextBufferForFrame(remaining);
     }
 
-    // Otherwise, create a composite buffer.
-    CompositeByteBuf frame = buffers.getFirst().alloc().compositeBuffer();
```

Comment:
@vanzin , I'm not sure why Netty underlying set a maximum number components (max size is Integer.MAX_VALUE), and the default value is only 16, this seems very small for consolidation.  Will it occurs other problem when there are too many small buffers under `compositeBuffer`? Is that why it will consolidate when the small buffer number reaches the `maxNumCompnent`?